docs/CloudFerry_Beginner_Demo_How-To.md

```diff
@@ -0,0 +1,218 @@
+(CF is written in Python), networking, databases, Unix text editors,
+large scale storage appliances, RESTful applications and, of course,
+substantial knowledge of and experience with OpenStack including at the
+command line.
+
+CloudFerry is a highly configurable tool and as such there are more ways
+to use it than can be described in a brief document. This demo will
+assist you in creating a pair of virtual OpenStack clouds, adding
+tenants and configuring resources for you to migrate. 
+
+In the real world it’s almost certain that since migration implys moving
+workloads from one cloud environment to another that you’ll need to
+perform a Condensation and Evacuation run prior to migration simply to
+free up enough hardware to pull it off. Those 2 topics are not covered
+in this document. 
+
+**Before You Begin**
+-----------------------
+This guide is meant to get a demo of CloudFerry going. Normally it would be installed
+on a host or VM which has network connectivity to the public IP's of both the source
+and destination clouds. It's advised that this migration host be a Debian/Ubuntu
+Linux distribution when doing real migrations or any significant testing.
+ 
+The instructions below use the "nfs" VM as the migration host but it could equally
+well be done directly from a Debian/Ubuntu or MacOS workstation. 
+
+It is not recommended to use CentOS as your migration host. 
+
+How-To: CloudFerry for Beginners
+================================
+**System Requirements**
+-----------------------
+
+1.  100GB free disk space as a minimum.
+2.  CPU power to spare 5 cores.
+3.  12GB free RAM (you should have 16GB total as a practical minimum)
+4.  Internet access (for git clone).
+5.  MacOS or Ubuntu Linux (or other Debian distribution) Operating
+    System
+
+**Setting Up CloudFerry Test Environment on MacOS**
+-----------------------
+1.  Download and install Vagrant http://www.vagrantup.com/downloads.html
+
+2.  Download and install VirtualBox https://www.virtualbox.org/wiki/Downloads
+    Open Terminal application and proceed.
+
+3.  [user@MacOS ~/]$ ```mkdir ~/git```
+
+4.  [user@MacOS ~/]$ ```cd ~/git```
+
+5.  [user@MacOS ~/git]$ ```git clone https://github.com/MirantisWorkloadMobility/CloudFerry.git```
+
+6.  [user@MacOS ~/git]$ ```cd ~/git/CloudFerry/devlab```
+
+7.  [user@MacOS ~/git/CloudFerry/devlab]$ ```vagrant up grizzly icehouse juno nfs```
+    
+    Open a new terminal window to watch logs.
+    Copy SSH keys around to make your life easier.
+
+8.  [user@MacOS ~/git/CloudFerry/devlab]$ ```cat ~/.ssh/id_rsa```
+
+9.  [user@MacOS ~/git/CloudFerry/devlab]$ ```cat ~/.ssh/id_rsa.pub```
+
+10. [user@MacOS ~/git/CloudFerry/devlab]$ ```vagrant ssh grizzly```
+
+    a.  [vagrant@grizzly ~/]$ ```vi ~/.ssh/id_rsa```
+        Paste the text from step 8 above.
+
+    b.  [vagrant@grizzly ~/]$ ```chmod 600 ~/.ssh/id_rsa```
+
+    c.  [vagrant@grizzly ~/]$ ```vi ~/.ssh/authorized_keys```
```

Comment:
All the extra steps with putting public keys inside VMs seems strange to me, because this should be already handled by Vagrantfile (see (provision/keyes.sh script)[https://github.com/MirantisWorkloadMobility/CloudFerry/blob/master/devlab/provision/keys.sh])...